teb_local_planner/src/optimal_planner.cpp

```diff
@@ -1037,6 +1034,13 @@ bool TebOptimalPlanner::hasDiverged() const
   // Grab the statistics of the final iteration
   const auto last_iter_stats = stats_vector.back();
 
+  // publish chi2
+  visualization_->publishChi2(last_iter_stats.chi2);
+
+  // Return if divergence detection is not active
+  if (!cfg_->recovery.divergence_detection_enable)
+    return false;
+
```

Comment:
put this here so we can still visualize chi2 even if it's currently disabled for pickup and putdown